### PR TITLE
Apple Pay Support component

### PR DESCRIPTION
**Pull request by swathikumar-shortcut**
This is a stand alone ApplePay handler, which returns the payment token in case of a successful transaction.
The token can then be forwarded to BE which has integration with Stripe, BrainTree or Adyen.
The UI of the component is not included in this module.

### CODE REVIEW

File: Sources/ShortcutFoundation/Core/Payment/ApplePayAuthorizationHandler.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
+//
+//  ApplePayAuthorizationHandler.swift
+//  ShortcutFoundation
+//
+//  Created by Swathi on 2022-02-13.
+//  Copyright © 2022 No more yellow notes AB. All rights reserved.
```
Comment (gabrielPeart) [inline Sources/ShortcutFoundation/Core/Payment/ApplePayAuthorizationHandler.swift:6]: All the copyrights in ShortcutFoundation should be for our company. Please, adjust it.
Maybe if something missed from others, we need to adjust.

File: Sources/ShortcutFoundation/Core/Payment/PaymentManager.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
+//
+//  PaymentManager.swift
+//  ShortcutFoundation
+//
+//  Created by Swathi on 2022-02-14.
+//  Copyright © 2022 No more yellow notes AB. All rights reserved.
```
Comment (gabrielPeart) [inline Sources/ShortcutFoundation/Core/Payment/PaymentManager.swift:6]: Same here

File: Sources/ShortcutFoundation/Core/Payment/PaymentModel.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
+//
+//  PaymentModel.swift
+//  ShortcutFoundation
+//
+//  Created by Swathi on 2022-02-14.
+//  Copyright © 2022 No more yellow notes AB. All rights reserved.
```
Comment (gabrielPeart) [inline Sources/ShortcutFoundation/Core/Payment/PaymentModel.swift:6]: Here

**Review by gabrielPeart — COMMENTED** [review]
Thank you. Great addition.

One necessary thing to do is to adjust the copyrights (vide my comments and all the other files are repeating the same pattern).

One reflection of mine is if we should include this part in Foundation or create a new dedicated module

**Comment by lemonandlime** [comment]
@swathikumar-shortcut I have fixed the failing circleCI test that shouldnt be there since we now use GitHubActions it also looks like your conversations are fixed so lets try to get it in there let me know when I should make another review

**Comment by gabrielPeart** [comment]
@swathikumar-shortcut @lemonandlime let's get it merged :)

Good job!

File: Makefile [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
+test:
+	swift test
+
 lint:
 	swiftlint --strict
 
-fix_lint:
-	swiftlint --fix
+fix_lint: 
+	swiftlint autocorrect
```
Comment (lemonandlime) [inline Makefile:11]: I think autocorrect is depricated

**Review by lemonandlime — APPROVED** [review]

File: Sources/ShortcutFoundation/Core/Payment/ApplePayAuthorizationHandler.swift (hunk 1 above)
Comment (swathikumar-shortcut) [reply Sources/ShortcutFoundation/Core/Payment/ApplePayAuthorizationHandler.swift]: Fixed

File: Sources/ShortcutFoundation/Core/Payment/PaymentModel.swift (hunk 3 above)
Comment (swathikumar-shortcut) [reply Sources/ShortcutFoundation/Core/Payment/PaymentModel.swift]: Fixed

File: Sources/ShortcutFoundation/Core/Payment/PaymentManager.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+import Foundation
+import PassKit
+
+protocol IPaymentManager {
+    var hasApplePaySupport: Bool { get }
+    func checkoutWithApplePay(for paymentModel: PaymentModel)
+}
+
+#if !os(macOS)
```
[the remark this answers is not in the document]
Comment (swathikumar-shortcut) [reply Sources/ShortcutFoundation/Core/Payment/PaymentManager.swift:17]: The macOS requires a delegate presentationWindow. But my idea was to make it work only for iOS. Please correct me if I am wrong

File: Makefile (hunk 4 above)
Comment (swathikumar-shortcut) [reply Makefile]: Updated to `swiftlint --fix --format`

**Review by gabrielPeart — APPROVED** [review]

**Pull request merged**